lib/appsignal/integrations/puma.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Appsignal
+  module Integrations
+    # @api private
+    module PumaServer
+      def lowlevel_error(error, env, response_status = 500)
+        response =
+          if method(:lowlevel_error).super_method.arity.abs == 3 # Puma >= 5
+            super
+          else # Puma <= 4
+            super(error, env)
+          end
+
+        unless PumaServerHelper.ignored_error?(error)
+          Appsignal.report_error(error) do |transaction|
+            Appsignal::Rack::ApplyRackRequest
+              .new(::Rack::Request.new(env))
+              .apply_to(transaction)
+            transaction.add_tags(
+              :reported_by_puma_lowlevel_error => true,
```

Comment:
Nit: since our global search only allows searching for tag _values_, `:reported_by => :puma_lowlevel_error` might make it easier to search for them.

(I think this tag is a great idea, by the way)